v8.12.0

ELundby45 opened the pull request:
curl v8.12.0

**Destination channel:** default

### Links

- [PKG-7004](https://anaconda.atlassian.net/browse/PKG-7004) 
- [Upstream repository](https://github.com/curl/curl/tree/curl-8_12_0)
- [Upstream changelog/diff](https://curl.se/ch/8.12.0.html)


### Explanation of changes:

- Port windows to cmake as msbuild has officially been [deprecated](https://github.com/curl/curl/blob/6ab1fa423bcc49a742b1cde2164ff981fdee38e8/docs/INSTALL-CMAKE.md#migrating-from-visual-studio-ide-project-files). 
- Not needed, but ported unix to cmake per @danpetry request. 
- Get rid of static libs
- Use same build flags for all archs
- Add tests to windows


[PKG-7004]: https://anaconda.atlassian.net/browse/PKG-7004?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

danpetry commented on recipe/bld.bat:
```diff
@@ -1,29 +1,21 @@
+    -DCMAKE_BUILD_TYPE=Release ^
+    -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
+    -DBUILD_SHARED_LIBS=ON ^
+    -DBUILD_STATIC_LIBS=ON ^
```
can we omit static libs?

danpetry commented on recipe/meta.yaml:
```diff
@@ -25,7 +27,7 @@ requirements:
     - openssl     {{ openssl }}  # [unix]
     - zlib        {{ zlib }}
-    - zstd        {{ zstd }}  # [unix]
+    - zstd        {{ zstd }}
```
I don't see zstd support added in bld.bat?

danpetry reviewed (commented):
to get parity across win and unix, you could consider:
- using cmake for unix too, and use the same options on both
- websockets support on win
- could disable ldap on win (explicitly)

ELundby45 commented:
> to get parity across win and unix, you could consider:
> - using cmake for unix too, and use the same options on both
> - websockets support on win
> - could disable ldap on win (explicitly)

We could add a new ticket for this. I think it's better to get the CVE addressed then the "nice to haves"

ELundby45 commented:
Ok @danpetry I've addressed the concerns mentioned. Note that windows is currently failing to sign the artifacts. It doesn't seem to be related to this package:

```
The provided client secret keys for app 'c4ef822b-d2af-4650-8040-16ef486b23d9' are expired.
```
